### PR TITLE
feat(datatype): new datatype display for desc & show create

### LINKED ISSUE #4914: New data type display for the user
**Summary**

All General-Purpose Data Types after refine(including datetime #4912 ):

| Syntax               | Size(Bytes)      | Min Value              | Max Value   | Description |
----------------------| --------- | ---------------------- | ----------- | -------|
| BOOLEAN              | 1 |                   | | Logical boolean (true/false)
| TINYINT              | 1 |  -128                  |  127 |
| SMALLINT             | 2 |  -32768                |  32767 |
| INT                  | 4 |  -2147483648           |  2147483647 |
| BIGINT               | 8 |  -9223372036854775808  |  9223372036854775807 |
|  FLOAT    | 4 |  -3.40282347e+38          | 3.40282347e+38 |
|  DOUBLE   | 8 |  -1.7976931348623157E+308 | 1.7976931348623157E+308 |
|  DATE      | 4 |  1925-01-01            | 2283-11-11                    | YYYY-MM-DD             |
|  DATETIME  | 8 |  1970-01-01 00:00:00   |  2283-11-11 23:59:59.99999999          | YYYY-MM-DD hh:mm:ss.ff    |

If the type is unsinged, append suffix `UNSIGNED`.

For example:

```
mysql> create table test(a TINYINT, b VARCHAR, c INT);
```

Now:
```
mysql> desc test;
+-------+--------+------+---------+
| Field | Type   | Null | Default |
+-------+--------+------+---------+
| a     | Int8   | NO   | 0       |
| b     | String | NO   |         |
| c     | Int32  | NO   | 0       |
+-------+--------+------+---------+


mysql> show create table test\G;
*************************** 1. row ***************************
       Table: test
Create Table: CREATE TABLE `test` (
  `a` Int8,
  `b` String,
  `c` Int32,
) ENGINE=FUSE
```

New:
```
mysql> desc test;
+-------+--------+------+---------+
| Field | Type   | Null | Default |
+-------+--------+------+---------+
| a     | TINYINT   | NO   | 0      |
| b     | VARCHAR | NO   |    |
| c     | INT  | NO   | 0   |
+-------+--------+------+---------+


mysql> show create table test\G;
*************************** 1. row ***************************
       Table: test
Create Table: CREATE TABLE `test` (
  `a` TINYINT,
  `b` VARCHAR,
  `c` INT,
) ENGINE=FUSE
```

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR:

new datatype display for desc & show create

 

## Changelog

 
- Improvement 

## Related Issues

Fixes #4914

### CODE REVIEW

File: common/datavalues/src/types/data_type.rs [hunk 1]
```diff
@@ -91,6 +91,11 @@ pub trait DataType: std::fmt::Debug + Sync + Send + DynClone {
 
     fn name(&self) -> &str;
 
+    /// Returns the name to display in the SQL describe
+    fn sql_name(&self) -> String {
+        self.name().to_lowercase()
```
Comment (BohuTANG) [inline common/datavalues/src/types/data_type.rs:96]: to_uppercase?

Comment (sundy-li) [reply common/datavalues/src/types/data_type.rs:96]: See the summary of this pr.

Comment (BohuTANG) [reply common/datavalues/src/types/data_type.rs:96]: I think this is a bad case for MySQL, his style is inconsistent.
Most databases is using uppercase, e.g. snowflake, duckdb

Our document changed to uppercase for the datatype now, realize that one of the benefits of uppercase is:

With uppercase:
```
CREATE TABLE t1(int INT);
```

With lowercase:
```
CREATE TABLE t1(int int); -- int is a field name or data type is trivial here
```

Comment (sundy-li) [reply common/datavalues/src/types/data_type.rs:96]: ok

**Review by BohuTANG — APPROVED** [review]
:rocket: 

I will change all the documents about the data type display after this PR merged.

**Pull request merged**